tests/plugin/test_http_proxy_plugins.py

```diff
@@ -24,21 +25,23 @@
 from proxy.http.proxy import HttpProxyPlugin
 from proxy.common.utils import build_http_request, bytes_, build_http_response
 from proxy.common.constants import PROXY_AGENT_HEADER_VALUE, DEFAULT_HTTP_PORT
-
 from proxy.plugin import ProposedRestApiPlugin, RedirectToCustomServerPlugin
 
 from .utils import get_plugin_by_test_name
 
+from ..test_assertions import Assertions
 
-class TestHttpProxyPluginExamples(unittest.TestCase):
 
-    @mock.patch('selectors.DefaultSelector')
-    @mock.patch('socket.fromfd')
-    def setUp(
-        self,
-        mock_fromfd: mock.Mock,
-        mock_selector: mock.Mock,
-    ) -> None:
+class TestHttpProxyPluginExamples(Assertions):
+
+    @pytest.fixture(autouse=True)   # type: ignore[misc]
+    def _setUp(self, request: Any, mocker: MockerFixture) -> None:
```

Comment:
It's unnatural to use Java-style in the context of Python / pytest. The previous name was a byproduct of the original `unittest` module in stdlib "copied" from Java. But nowadays, unless you're forced to use `unittest` directly, don't use camelCase for methods/functions.

Reply:
Will address these changes separately as this residual effect is spread across all the test files.